source/ast/Scope.cpp

```diff
@@ -1165,6 +1177,26 @@ void Scope::elaborate() const {
         }
     }
 
+    // Unnamed generate blocks must be handled last to detect genblk name collisions
+    for (auto symbol : unnamedGenblks) {
+        auto updateName = [&](Symbol* symbol, const std::string& externalName) {
+            char* mem = (char*)getCompilation().allocate(externalName.size(), 1);
+            memcpy(mem, externalName.data(), externalName.size());
+            symbol->name = std::string_view(mem, externalName.size());
+            nameMapEmplace(symbol, true);
+        };
+        if (auto block = symbol->as_if<GenerateBlockSymbol>()) {
+            if (block->name.empty())
```

Comment:
The name is always empty here since you checked above before adding to the list, so you don't need to check again.